### PR TITLE
[BE][Easy] sort `__all__` in `torch/__init__.py`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127705
* #127704
* #128342
* #127707
* #127706
* #127710
* __->__ #127709
* #127708
* #127703

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -50,28 +50,80 @@ def _running_with_deploy():
 
 
 __all__ = [
```
Comment (Skylion007) [inline torch/__init__.py:52]: ruff rule RUF022 will take care of this for us once it's out of preview.

File: torch/__init__.py [hunk 2]
```diff
@@ -50,28 +50,80 @@ def _running_with_deploy():
...
+    "sym_min",
+    "sym_not",
+    "typename",
+    "unravel_index",
+    "use_deterministic_algorithms",
+    "vmap",
 ]
 
+# Please keep this list sorted
+assert __all__ == sorted(__all__)
```
Comment (Skylion007) [inline torch/__init__.py:125]: We can just use RUF022 for this.

File: torch/__init__.py [hunk 3]
```diff
@@ -381,35 +439,35 @@ def __bool__(self):
     # Magic methods installed by torch.fx.experimental.sym_node
 
     def __eq__(self, other: object) -> builtins.bool:
-        raise AssertionError("type stub not overridden")
+        raise TypeError("type stub not overridden")
```
Comment (Skylion007) [inline torch/__init__.py:442]: Should these be NotImplementedError s?

File: torch/__init__.py (hunk 1 above)
Comment (XuehaiPan) [reply torch/__init__.py:52]: Not sure if we'd like to enable RUF022 globally.

Comment (Skylion007) [reply torch/__init__.py:52]: Why? Sorting the `__all__` list seems harmless.

File: torch/__init__.py (hunk 3 above)
Comment (XuehaiPan) [reply torch/__init__.py]: Updated.

File: torch/__init__.py (hunk 1 above)
Comment (XuehaiPan) [reply torch/__init__.py:52]: Sometimes, we add comments in `__all__` for better readability.

```python
__all__ = [
    # Module APIs
    "AaaModule",
    "BbbModule",
    "CccModule",
    # Dataset APIs
    "AaaDataset",
    "BbbDataset",
    "CccDataset",
]
```

After sorting:

```python
__all__ = [
    "AaaModule",
    "AaaDataset",
    "BbbModule",
    "BbbDataset",
    "CccModule",
    "CccDataset",
]
```

File: torch/__init__.py [hunk 4]
```diff
@@ -398,55 +456,55 @@ def __rpow__(self, other):
             return sym_float(self).__rpow__(sym_float(other))
 
     def __eq__(self, other: object) -> builtins.bool:
-        raise AssertionError("type stub not overridden")
+        raise NotImplementedError("type stub not overridden")
```
Comment (ezyang) [inline torch/__init__.py:459]: AssertionError is right here. We have some initialization code elsewhere which is supposed to override all of these. If there is a problem with that code, this is a framework bug and we should assert fail

**Review by ezyang — APPROVED** [review]
okey dokey

**Review by ezyang — APPROVED** [review]
okey dokey

File: torch/__init__.py (hunk 4 above)
Comment (ezyang) [reply torch/__init__.py:459]: NotImplementedError implies that someday someone in the framework will implement it. That's 100% not the intent here, if you hit this code, it is a BUG. These stubs exist purely to improve type checking.

Comment (XuehaiPan) [reply torch/__init__.py:459]: > NotImplementedError implies that someday someone in the framework will implement it.

I see. Seems that `RuntimeError` or `TypeError` would be better.

**Pull request closed without merging**